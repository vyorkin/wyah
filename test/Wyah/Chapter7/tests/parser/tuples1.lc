-- Tuples

let fst p = p T;
let snd p = p F;
let pair a b f = f a b;
