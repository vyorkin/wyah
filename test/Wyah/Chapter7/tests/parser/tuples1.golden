Program [Decl "fst" (ELam (Var "p") (EApp (EVar (Var "p")) (EVar (Var "T")))),Decl "snd" (ELam (Var "p") (EApp (EVar (Var "p")) (EVar (Var "F")))),Decl "pair" (ELam (Var "a") (ELam (Var "b") (ELam (Var "f") (EApp (EApp (EVar (Var "f")) (EVar (Var "a"))) (EVar (Var "b"))))))]